a1/src/main/java/ro/utcn/is/a1/controller/ConsoleController.java

```diff
@@ -0,0 +1,324 @@
+        String title = scanner.next().trim();
+        int authorId = user.getId();
+        print("Text:");
+        String text = scanner.next().trim();
+        Date dateText = new Date();
+
+        Question question= questionManagementService.insert(title, authorId, text, dateText);
+        print("Created question: "+ title + ".");
+    }
+
+    private void handleAddT() throws Exception{
+        print("Tag Text:");
+        String tagText = scanner.next().trim();
+
+        Tag tag = tagManagementService.addTag(tagText);
+        print("Created tag: " + tag + ".");
+    }
+
+    //-------------- UPDATES
+    private void handleUpdateUsername() {
+        print("User ID:");
+        int id = scanner.nextInt();
+        print("Username:");
+        String username = scanner.next().trim();
+        userManagementService.updateUsername(id, username);
+    }
+
+    private void handleUpdateQ(User user) throws Exception{
+        print("Question ID:");
+        int id = scanner.nextInt();
+        print("Title:");
+        String title = scanner.next().trim();
+        //print("Author:");
+        int authorId = user.getId();
+        print("Text:");
+        String text = scanner.next().trim();
+        //print("Date:");
+        Date dateText = new Date();
+
+        questionManagementService.update(id, title, authorId, text, dateText);
+    }
+
+    private void handleUpdateT() {
+        print("Tag ID:");
+        int id = scanner.nextInt();
+        print("Tag Text:");
+        String tagText = scanner.next().trim();
+        tagManagementService.updateTag(id, tagText);
+    }
+
+    //-------------- REMOVES
+    private void handleRemoveU() {
+        print("User ID:");
+        int id = scanner.nextInt();
+        userManagementService.removeUser(id);
+    }
+
+    private void handleRemoveQ(User user) {
+        print("Question ID:");
+        int id = scanner.nextInt();
+        List<Question> qList = questionManagementService.ListQuestionByUser(user);
+        for(Question q :qList){
+            if(q.getId()== id){
+                questionManagementService.remove(id);
+            }
+        }
+
+    }
+
+    private void handleRemoveT() {
+        print("Tag ID:");
+        int id = scanner.nextInt();
+        tagManagementService.removeTag(id);
+    }
+
+
+    //-------------- PRINT
+    private void print(String value) {
+        System.out.println(value);
+    }
+
+    private void printQuestionsByTag(){
+        print("filter Tag: ");
+        String tagText = scanner.next().trim();
+        Tag t= tagManagementService.getTag(tagText);
+
+            questionTagManagementService.findByTagId(t.getId()).forEach(s-> print(questionManagementService.findById(s).toString()));
+
+
+    }
+
+    private void qTextSearch(){
+        print("Search title: ");
+        String title = scanner.next().trim();
+        List<Question> questions = questionManagementService.listQuestions();
+
+        for(Question q: questions ){
```

Comment:
This is business logic or even persistence logic! It should either reside in the service or the persistence layer.